Introduce Data handler (#2719)
freqtrade/freqtrade: docs/data-download.md

                               [--erase] [--data-format {json,jsongz}] [--data-format-trades {json,jsongz}]

optional arguments:
  -h, --help            show this help message and exit
  -p PAIRS [PAIRS ...], --pairs PAIRS [PAIRS ...]
                        Show profits for only these pairs. Pairs are space-separated.
  --pairs-file FILE     File containing a list of pairs to download.
  --days INT            Download data for given number of days.
  --dl-trades           Download trades instead of OHLCV data. The bot will resample trades to the desired timeframe as specified as
                        --timeframes/-t.
  --exchange EXCHANGE   Exchange name (default: `bittrex`). Only valid if no config is provided.
  -t {1m,3m,5m,15m,30m,1h,2h,4h,6h,8h,12h,1d,3d,1w} [{1m,3m,5m,15m,30m,1h,2h,4h,6h,8h,12h,1d,3d,1w} ...], --timeframes {1m,3m,5m,15m,30m,1h,2h,4h,6h,8h,12h,1d,3d,1w} [{1m,3m,5m,15m,30m,1h,2h,4h,6h,8h,12h,1d,3d,1w} ...]
                        Specify which tickers to download. Space-separated list. Default: `1m 5m`.
  --erase               Clean all existing data for the selected exchange/pairs/timeframes.

>   --data-format {json,jsongz}

Reviewer: this should probably better be called `--data-format-ohlcv`, to have it aligned with `dataformat_ohlcv` setting...

                        Storage format for downloaded ohlcv data. (default: `json`).
  --data-format-trades {json,jsongz}
                        Storage format for downloaded trades data. (default: `jsongz`).

Common arguments:
  -v, --verbose         Verbose mode (-vv for more, -vvv to get all messages).
  --logfile FILE        Log to the file specified. Special values are: 'syslog', 'journald'. See the documentation for more details.
  -V, --version         show program's version number and exit
  -c PATH, --config PATH
                        Specify configuration file (default: `config.json`). Multiple --config options may be used. Can be set to `-`
                        to read config from stdin.
  -d PATH, --datadir PATH
                        Path to directory with historical backtesting data.
  --userdir PATH, --user-data-dir PATH
                        Path to userdata directory.
```

### Data format

> Freqtrade currently supports 2 dataformats, `json` and `jsongz`, a zipped version of json files.

Reviewer: ```suggestion
Freqtrade currently supports 2 dataformats, `json` (plain "text" json files) and `jsongz` (a gzipped version of json files).
```

> By default, OHLCV data is stored as json data, while trades data is stored as `jsongz` data.

Reviewer: ```suggestion
By default, OHLCV data is stored as `json` data, while trades data is stored as `jsongz` data.
```

If the default dataformat has been changed during download, then the keys `dataformat_ohlcv` and `dataformat_trades` in the configuration file need to be adjusted to the selected dataformat as well.

!!! Note
    You can convert between data-formats using the [convert-data](#subcommand-convert-data) and [convert-trade-data](#subcommand-convert-trade-data) methods.

#### Subcommand convert data

```
usage: freqtrade convert-data [-h] [-v] [--logfile FILE] [-V] [-c PATH]
                              [-d PATH] [--userdir PATH]
                              [-p PAIRS [PAIRS ...]] --format-from
                              {json,jsongz} --format-to {json,jsongz}
                              [--erase]
                              [-t {1m,3m,5m,15m,30m,1h,2h,4h,6h,8h,12h,1d,3d,1w} [{1m,3m,5m,15m,30m,1h,2h,4h,6h,8h,12h,1d,3d,1w} ...]]

optional arguments:
  -h, --help            show this help message and exit
  -p PAIRS [PAIRS ...], --pairs PAIRS [PAIRS ...]
                        Show profits for only these pairs. Pairs are space-
                        separated.
  --format-from {json,jsongz}

>                         Source format for data conversation.

Reviewer: "data conversion"
pls check other appearances

                        exchange/pairs/timeframes.
  -t {1m,3m,5m,15m,30m,1h,2h,4h,6h,8h,12h,1d,3d,1w} [{1m,3m,5m,15m,30m,1h,2h,4h,6h,8h,12h,1d,3d,1w} ...], --timeframes {1m,3m,5m,15m,30m,1h,2h,4h,6h,8h,12h,1d,3d,1w} [{1m,3m,5m,15m,30m,1h,2h,4h,6h,8h,12h,1d,3d,1w} ...]
                        Specify which tickers to download. Space-separated
                        list. Default: `1m 5m`.

Common arguments:
  -v, --verbose         Verbose mode (-vv for more, -vvv to get all messages).
  --logfile FILE        Log to the file specified. Special values are:
                        'syslog', 'journald'. See the documentation for more
                        details.
  -V, --version         show program's version number and exit
  -c PATH, --config PATH
                        Specify configuration file (default: `config.json`).
                        Multiple --config options may be used. Can be set to
                        `-` to read config from stdin.
  -d PATH, --datadir PATH
                        Path to directory with historical backtesting data.
  --userdir PATH, --user-data-dir PATH
                        Path to userdata directory.
```

##### Example converting data

> The following command will convert all Candle data available in `~/.freqtrade/data/binance` from json to jsongz, saving diskspace in the process.

Reviewer: ```suggestion
The following command will convert all ohlcv (candle) data available in `~/.freqtrade/data/binance` from json to jsongz, saving diskspace in the process.
```

> It'll also remove source files (`--erase` parameter).

Reviewer: ```suggestion
It'll also remove original json data files (`--erase` parameter).
```

                        separated.
  --format-from {json,jsongz}
                        Source format for data conversion.
  --format-to {json,jsongz}
                        Destination format for data conversion.
  --erase               Clean all existing data for the selected
                        exchange/pairs/timeframes.

Common arguments:
  -v, --verbose         Verbose mode (-vv for more, -vvv to get all messages).
  --logfile FILE        Log to the file specified. Special values are:
                        'syslog', 'journald'. See the documentation for more
                        details.
  -V, --version         show program's version number and exit
  -c PATH, --config PATH
                        Specify configuration file (default: `config.json`).
                        Multiple --config options may be used. Can be set to
                        `-` to read config from stdin.
  -d PATH, --datadir PATH
                        Path to directory with historical backtesting data.
  --userdir PATH, --user-data-dir PATH
                        Path to userdata directory.
```

##### Example converting trades

> The following command will convert all available trade-data in `~/.freqtrade/data/kraken` from json to jsongz, saving diskspace in the process.

Reviewer: ```suggestion
The following command will convert all available trade-data in `~/.freqtrade/data/kraken` from jsongz to json.
```

> It'll also remove source files (`--erase` parameter).

Reviewer: ```suggestion
It'll also remove original data files (`--erase` parameter).
```
